pandas/_libs/window_indexer.pyx

```diff
@@ -0,0 +1,167 @@
+# cython: boundscheck=False, wraparound=False, cdivision=True
+
+import numpy as np
+from numpy cimport ndarray, int64_t
+
+# ----------------------------------------------------------------------
+# The indexer objects for rolling
+# These define start/end indexers to compute offsets
+
+
```

Comment:
can we use a base class that defined get_window_bounds as NotImplementedError

Reply:
Was planning on doing this in a follow up PR.